### PR TITLE
[disasters] read disaster event map data from saved jsons

### PR DESCRIPTION
**Pull request by chejennifer**
- add new flask endpoint that reads disaster event map data from saved jsons
- add url hash param that will read data from cache if "&cache=1" is set

### CODE REVIEW

File: server/lib/disaster_dashboard.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      }
+  """
+  result = {}
+  for event_type in EVENT_TYPES:
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(gcs_bucket)
+    file_name = re.sub('(?!^)([A-Z]+)', r'_\1', event_type).lower() + ".json"
+    blob = bucket.get_blob(DISASTER_DATA_FOLDER + file_name)
+    if not blob:
+      continue
```
Comment (beets) [inline server/lib/disaster_dashboard.py:61]: nit: log a comment if there's a missing bucket?

File: server/routes/api/disaster_api.py [hunk 2]
```diff
@@ -49,21 +50,134 @@ def event_date_range():
...
+  if place != EARTH_DCID and not place in event.get("affectedPlaces", []):
+    # event does not affect the place of interest
+    return False
+  if not filter_prop or not filter_prop in event:
+    # no props to filter by or event doesn't contain the prop information
+    return True
+  prop_val = event[filter_prop]
+  if len(prop_val) < len(filter_unit):
+    # value for the property to filter by is empty
+    return True
```
Comment (beets) [inline server/routes/api/disaster_api.py:67]: we should return false here?

File: server/routes/api/disaster_api.py [hunk 3]
```diff
@@ -49,21 +50,134 @@ def event_date_range():
...
+  val_string = prop_val[len(filter_unit):].strip()
+  try:
+    val = float(val_string)
```
Comment (beets) [inline server/routes/api/disaster_api.py:70]: might need to `trim()` the string before float conversion?

File: server/routes/api/disaster_api.py [hunk 4]
```diff
@@ -49,21 +50,134 @@ def event_date_range():
...
+    return val <= filter_upper_limit and val >= filter_lower_limit
+  except:
+    # can't parse the value to filter by into a float
+    return True
```
Comment (beets) [inline server/routes/api/disaster_api.py:74]: Could we also log here so we can debug these as they come up?

File: server/routes/api/disaster_api.py [hunk 5]
```diff
@@ -49,21 +50,134 @@ def event_date_range():
...
+  date = request.args.get('date', '')
+  if not date:
+    return "error: must provide a date field", 400
+  place = request.args.get('place', '')
+  if not place:
+    return "error: must provide a place field", 400
+  filter_prop = request.args.get('filterProp', '')
+  filter_unit = request.args.get('filterUnit', '')
+  filter_upper_limit = float(request.args.get('filterUpperLimit', '0'))
+  filter_lower_limit = float(request.args.get('filterLowerLimit', '0'))
```
Comment (beets) [inline server/routes/api/disaster_api.py:121]: should we default to sys.float_info.max or  float("inf") and -float("inf")

File: server/routes/api/disaster_api.py [hunk 6]
```diff
@@ -49,21 +50,134 @@ def event_date_range():
...
+      }
+      for eventKey in event.keys():
+        if eventKey in EVENT_POINT_KEYS:
+          continue
+        event_formatted["propVals"][eventKey] = {"vals": [event[eventKey]]}
+      event_points.append(event_formatted)
+  result = {}
+  if event_points:
+    result = {"eventCollection": {"events": event_points, "provenanceInfo": {}}}
+  return json.dumps(result), 200
```
Comment (beets) [inline server/routes/api/disaster_api.py:148]: suggest returning this as Response(json.dumps(result), 200, "application/json") so you can format these in the inspector.

**Review by beets — COMMENTED** [review]
feel free to fix these as follow ups in the interest of time.

File: server/configmodule.py [hunk 7]
```diff
@@ -124,6 +124,7 @@ class LocalBaseConfig(Config):
   AI_CONFIG_PATH = os.path.abspath(
       os.path.join(os.path.curdir, '..', 'deploy/overlays/local/ai.yaml'))
   SCHEME = 'http'
+  GCS_BUCKET = 'datcom-website-autopush-resources'
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply server/configmodule.py:127]: the stanford instances already have their own GCS_BUCKET

File: server/routes/api/disaster_api.py (hunk 2 above)
Comment (chejennifer) [reply server/routes/api/disaster_api.py:67]: in mixer, if the prop isn't found or can't be parsed, we keep the event .. not sure if we want this logic to be different from mixer logic?

File: server/routes/api/disaster_api.py (hunk 3 above)
Comment (chejennifer) [reply server/routes/api/disaster_api.py:70]: I used `.strip()`, I think that's the same thing?

File: server/routes/api/disaster_api.py (hunk 5 above)
Comment (chejennifer) [reply server/routes/api/disaster_api.py:121]: we can, but defaulting to 0 makes the logic easier & the only time we will have to use default is if we get a filter_prop but no upper and lower limits which shouldn't happen .. I can make the change if you feel strongly

File: server/lib/disaster_dashboard.py (hunk 1 above)
Comment (chejennifer) [reply server/lib/disaster_dashboard.py:63]: done

File: server/routes/api/disaster_api.py (hunk 4 above)
Comment (chejennifer) [reply server/routes/api/disaster_api.py:78]: done

File: server/routes/api/disaster_api.py (hunk 6 above)
Comment (chejennifer) [reply server/routes/api/disaster_api.py]: done

**Comment by chejennifer** [comment]
thanks for the review! Made some of the suggested changes & left comments for the rest. If more changes are needed, I can address them in a followup

**Review by beets — APPROVED** [review]

**Comment by beets** [comment]
thanks - sorry for late approval!

**Pull request merged**